src/popup/router/routes.ts

```diff
@@ -113,7 +115,15 @@ export const routes: WalletAppRouteConfig[] = [
       },
       {
         path: 'account-details/',
-        component: AccountDetails,
+        /**
+         * New way of loading pages specific to the protocol.
+         * `componentName` prop is required.
+         * TODO:
+         *   1. Use the same method with all child components
+         *   2. Remove this comment
+         */
+        component: ProtocolSpecificView,
+        props: { componentName: PROTOCOL_VIEW_ACCOUNT_DETAILS },
```

Comment:
It may be that ProtocolSpecificView is overengineered. Have you considered doing separate components like ViewAccountDetailsCommon which render a blockchain-specific component? For example, write a function that would generate a component like ViewAccountDetailsCommon by child components.
```js
// ViewAccountDetailsCommon.js
export default generateProtocolSpecificView({
  aeternity: () => import('...'),
  bitcoin: () => import('...'),
  ...
});
```
This way would be simpler to track dependencies: if ViewAccountDetailsCommon is used somewhere, then all child components are used as well. Otherwise, you need to track the usage of ProtocolSpecificView with a specific prop value. Also, ViewAccountDetailsCommon would be a bit clearer to read.

Reply:
Yes, this was my first thought to do it this way. But during the implementation I found out that it will be bad for the scalability. This PR covers 2 views, but there will be more (10-15). With every view you would need to create one `Common` component, one boilerplate and one for every protocol. With 6 protocols it will be 8 components. So it's 8*15 = 120 components.

Also every common component would have to have the set of imports for every protocol. So whenever you add/remove protocol you would need to edit every common component.

My goal was to make the protocol installation as easy as possible, so I wanted to encapsulate all the imports inside of the protocol directory.

Reply:
So, you would have 15 Common components that would combine 120 specific components. Sounds clear. The only thing to propose is to define all common components in a single file. It would take less than 200loc. It doesn't seem to be bad because all these components work the same, except they combine different child components.

Reply:
It's rather subjective but I'm opting for the current variant. The `ProtocolViewsConfig` type covers the required components for each of the protocols. And the one component that we have right now implements it nicely. But I'm open for the opinion from the rest of the team.
@pstachel , @subhod-i , @CedrikNikita , @konradodo , @ifaouibadi ?

Reply:
I'm open to both solutions - I was also thinking about what @davidyuk proposed here but maintaining it in the future may be a curse for us. On the other hand it will provide us clear structure of components and one entry point for each view. I feel that @peronczyk solution is cleaner (despite the fact that I'm not so keen on using component names as props) and immediately will show us that we need to deal with the protocol specific view.
What do you think of mixing both solutions - to have dynamic imports based on the protocol to prevent maintaining the if-s or switches inside base components and to use it inside mentioned component templates?

Reply:
@pstachel can you give some code example of your proposal? I'm not sure if I understand it correctly.

Reply:
Sure, let's do it with an example of account details. 

We would have the base component AccountDetails.vue
```
<template>
  <ProtocolSpecificView :viewComponentName="PROTOCOL_VIEW_ACCOUNT_DETAILS" />
</template>
```

Reply:
But on a second thought I'm more convinced on your proposal as we wouldn't use any views directly in the templates, so we won't see the usage of `<ProtocolSpecificView />` tag

Reply:
To unblock further development I decided to create separate spike for this topic: https://aeternity-blockchain.atlassian.net/browse/SW-1100